src/midi/midi_fluidsynth.cpp

```diff
@@ -0,0 +1,245 @@
+	str_prop->Set_help("Fluidsynth chorus depth.");
+}
+
+bool MidiHandlerFluidsynth::Open(MAYBE_UNUSED const char *conf)
+{
+	LocaleGuard locale_guard;
+
+	Close();
+
+	auto *section = static_cast<Section_prop *>(control->GetSection("midi"));
+
+	fluid_settings_ptr_t settings(new_fluid_settings(), delete_fluid_settings);
+
+	auto get_double = [section](const char *const propname) {
+		try {
+			return std::stod(section->Get_string(propname));
+		} catch (const std::exception &e) {
+			/*
+			log_cb(RETRO_LOG_WARN,
+			       "[dosbox] error reading floating point '%s' "
+			       "conf setting: %s\n",
+			       e.what());
+		       */
+			return 0.0;
+		}
+	};
+
+	fluid_settings_setnum(settings.get(), "synth.sample-rate",
+	                      section->Get_int("fluid.samplerate"));
+	fluid_settings_setnum(settings.get(), "synth.gain", get_double("fluid.gain"));
+	fluid_settings_setint(settings.get(), "synth.polyphony",
+	                      section->Get_int("fluid.polyphony"));
+	fluid_settings_setint(settings.get(), "synth.cpu-cores",
+	                      section->Get_int("fluid.cores"));
+
+	fluid_settings_setint(settings.get(), "synth.reverb.active",
+	                      section->Get_bool("fluid.reverb"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.room-size",
+	                      get_double("fluid.reverb.roomsize"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.damp",
+	                      get_double("fluid.reverb.damping"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.width",
+	                      get_double("fluid.reverb.width"));
+	fluid_settings_setnum(settings.get(), "synth.reverb.level",
+	                      get_double("fluid.reverb.level"));
+
+	fluid_settings_setint(settings.get(), "synth.chorus.active",
+	                      section->Get_bool("fluid.chorus"));
+	fluid_settings_setint(settings.get(), "synth.chorus.nr",
+	                      section->Get_int("fluid.chorus.number"));
+	fluid_settings_setnum(settings.get(), "", get_double("fluid.chorus.level"));
+	fluid_settings_setnum(settings.get(), "synth.chorus.speed",
+	                      get_double("fluid.chorus.speed"));
+	fluid_settings_setnum(settings.get(), "synth.chorus.depth",
+	                      get_double("fluid.chorus.depth"));
+
+	fsynth_ptr_t fluid_synth(new_fluid_synth(settings.get()), delete_fluid_synth);
+	if (!fluid_synth) {
+		// log_cb(RETRO_LOG_WARN, "[dosbox] Error creating fluidsynth synthesiser\n");
+		return false;
+	}
+
+	std::string soundfont = section->Get_string("fluid.soundfont");
+	if (!soundfont.empty() && fluid_synth_sfcount(fluid_synth.get()) == 0) {
+		fluid_synth_sfload(fluid_synth.get(), soundfont.data(), true);
+	}
+
+	mixer_channel_ptr_t mixer_channel(
+	        MIXER_AddChannel(mixerCallback,
+	                         section->Get_int("fluid.samplerate"), "FSYNTH"),
+	        MIXER_DelChannel);
+	mixer_channel->Enable(true);
```

Comment:
Ideally mixer channels should be left disabled until the game triggers / enables / plays.  So in this case, I wonder if there's a MIDI register that the games write to, before midi starts to play? If so, we could enable the mixer there. 
Otherwise, the mixer spins at full speed (processing ~88000 zero ints through a pretty long function); and I suspect a lot of people will just leave fluid turned on, but only a subset of games will use it. So performance will be hurt when they need max-cycles.

Reply:
> I wonder if there's a MIDI register that the games writes to, before midi starts to play?

I don't know.

> and I suspect a lot of people will just leave fluid turned on

On Windows and macOS this shouldn't be a problem (native MIDI sequencers should be picked by default there, so fluid will be off), it is a problem on Linux and other OSes though (as e.g. on Linux ALSA port detection is bogus and Fluid is picked as a fallback almost always).

ALSA bug on Linux is its own thing, but I will investigate if we can prevent mixer from doing pointless work if there's no output originating from FSYNTH channel.

Also,  I don't know if enabling channel is required for it, but users need to be able to change volumne for FSYNTH channel using `mixer` command.

Reply:
>  I don't know if the channel needs to be enabled in order to change the volume of the FSYNTH channel using `mixer` command.

They're separate; but the channel does need to exist in the first place (because it's name is used in the mixer to find it).  the mixer command just overwrites the channel's public: volmain[] floats. Ie: 1.0 to 2.0, if you set mixer SB 200, for example.